### ISSUE TITLE
Review JOSS - BR

### ISSUE THREAD

**Issue by RiesBen** [GOMC-WSU/MoSDeF-dihedral-fit#100]
JOSS Review: https://github.com/openjournals/joss-reviews/issues/6970
------------------------------------------------------------------------------------

Hi I looked through you submission. In general the repository looks good to me! I installed the package and tested the unit tests. For the submission I have the following comments & suggestions:

- Repository
  - Readme:
    - [x] Install Section should explain the steps, as there is an additional program installed. This is important for totally new users.
    - [x] Suggestion: Indicate, that the box after the code example is the output.
  - [x] Setup.py: Authors and Emails missing
  - [x] Environment.yml: Could it be that GOMC is missing? If you state here in the docs, it is an requirement (https://mosdef-dihedral-fit.readthedocs.io/en/latest/getting_started/quick_start/quick_start.html#), then it should be part of the environment. (could also make binary detection easier)
   - [x] Documentation: Repository Doc strings - API Rcference Docs missing? (can be generated with `sphinx-apidoc`)
   - [x] file_read_and_write.py: If I see it correctly this one contains unused imports.
   - [x] License File: if you change the formatting, then GitHub can recognize the license file (e.g.: https://github.com/OpenFreeEnergy/kartograf/blob/main/LICENSE)
- Paper: 
  - [x]  Line 12-24: Introduction
    - [x] line 12: MD and MC are just two examples, you could indicate this with e.g.
    - [x] Line 12-14: I find the categorization of theoretical, simulation and experimental scientific methods a bit confusing and would like to offer the following view: "There are theoretical and experimental methods, in the theoretical regime we can distinguish between analytical and stochastic methods. Stochastic methods can be for example simulation approaches. In the regime of MM the size of the phase space of more complex systems makes analytical methods very hard to calculate certain properties. However if well converged simulations allow to observe the relevant parts of the phase space and therefore give important estimates of certain properties. ..."
    - [x] Line 15: what are conventional methods here? I'm not a native speaker, but I have the impression that difficult and unattainable is saying kinda the same? (too expensive as well)
    - [x] Line 19: QM parameters are nice if used in forcefields, but they are not necessarily a requirement (GROMOS - FF, etc.).
    - [x]  Line 12-24: You should back up your text with literature, here.
  - [x] Line 39 - 40:  I'm not sure I understand why turning off other contributions makes the dihedral fit more accurate, can you elaborate please?
  - [x] Line 87-88: Your claim here needs to be supported by literature here.
  - [x] Line 100-103: I would like to suggest for the equations to use physical meaningful symbols as targets quantities (U or V). Examples: V_{dihedral}^{OPLS}(\theta) (instead of OPLS_{Energy}), V_{dihedral}^{RB}(\psi), V_{dihedral}^{Periodic}(\theta)
  - A general remark:  I think the Introduction and statement of need are a bit repetitive, maybe you could slim down the paper a bit?

**Comment by bc118** [comment]
Hello!

Thank you for your review and comments!  We will complete all of these that are feasible to completed. We started a issue checklist with all the reviewer comments, so we can address them and write our teams notes here https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/issues/101.   

**I would like to request clarification on a few of your comments, so we can appropriately address them:**



_**18.**_ [ ]  **Line 19: QM parameters are nice if used in forcefields, but they are not necessarily a requirement (GROMOS - FF, etc.).** 

@bc118 added : I believe this line needs addressed "The `MoSDeF-Dihedral-Fit` software uses the QM and MM data to fit the dihedral for the specific force field, fitting the constants for the OPLS dihedral equation form with the correct combining rules and 1-4 scaling factors, as specified in the MoSDeF XML force field file. "  

I am not sure what you are requesting or suggesting here?  Just adding something here that these fits are not required for all FFs, (i.e., GROMOS - FF, etc.)?   @RiesBen Can you confirm or clarify this?



**_20._** [ ]  **Line 39 - 40:  I'm not sure I understand why turning off other contributions makes the dihedral fit more accurate, can you elaborate please?**  

@RiesBen Can you please confirm this is correct statement?  "The `MoSDeF-dihedral-fit` software accomplishes all this and automatically accounts for any of the common combining rules, allows the zeroing out any other dihedrals in the molecule (i.e., allowing a better fit and user control by setting the other dihedral energies to zero), and accounts for any 1-4 scaling factors specified via the MoSDeF XML files [Cummings:2021; Summers:2020; GMSO:2019; forcefield-utilities:2022], which contain the force fields. "

**If this is what you are talking about, I can try and find literature or add the example write-up below to the paper if you want.  Note: I did test this for the best fit, which I plan on publishing in another article.*
- The case of the dihedrals around the carboxylic acid CCCOOH (different oxygen atom classes --> OH=Oxygen with H, O=Oxygen without H).  This means there are 2 dihedrals in the same dihedral rotation (C-C-C-O and C-C-C-OH) that need to be fit at the same time or in parts that produce an accurate representation when the are combined together.  
-  Fitting one separately and  then the other one as a combined carboxylic acid system does not produce accurately fit data, as you can not properly fit these forces that are different together than they are separately.  I tried fit them separately and then together, by fitting one or the other first (i.e., the alcholol or keytone dihedral part), then fitting the remaining as combined on the carboxylic acid (CCCOOH).  However, getting a good fit with the power series <=4-5 is not easy to do this way.  
-- Zeroing out one of the dihedrals and fitting the QM data as 1 fit, produced much more accurate fits, while accounting for the interactions of both, since they are in the same dihedral rotation.  When these atom classes are always together in a structure, like carboxylic acid or amides, this is done frequently in OPLS fits and likely others
-  _**I discovered that zeroing 1 dihedral that is in the same rotational angle (just offset) was the optimal solution when I looked at the previous OPLS force field fits for amides to determine the best approach for this.**_  



**_21._** [ ]  **Line 87-88: Your claim here needs to be supported by literature here.**  @bc118 is not sure what line this is, as it is not matching what is on GitHub, which is "
$$OPLS_{Energy} = \frac{f_0}{2}$$" on lines 87-88 .  

 @RiesBen Can you please confirm that it is the lines I reference below, and which specific ones?    **

**mosdef-dihedral-fit thinks the lines below are the lines you are referring to.**  
 "Commonly used force field dihedrals (Example: OPLS) were originally fit when the QM simulations used to fit the dihedrals were computationally prohibitive.  
Due to these limitations, scientists assumed that the dihedral fits were transferable with all the atom classes in the dihedral fit; however, this is not always an accurate assumption.  Some of the dihedrals were only fit to the first minimum and not the entire dihedral landscape, which can lead to errors in the predicted molecular conformations.  These prior assumptions in the dihedral fits may also lead to problems in reproducibility in modified force fields.  "

- @RiesBen If this is what you are referring to, I do have proof from using this dihedral fit software that all the current atom classes (at least in OPLS) do not always work well to fit the dihedrals (see proof I have below, which I can provide to you if no matching literature can be found).. 

**Proof is listed here:** 
 @bc118  Has a specific newly fit dihedral fit example where 0-1 alkyl R-groups at the end of a dihedral match the existing OPLS dihedral (less the different QM theory accuracy used).  When fitting with 2-3 alkyl R-groups, there is a higher energy maximum where the original dihedral fit has a minimum, which would change the angle somewhat significantly if the 2-3 alkyl R-groups are used in the molecule, making the original or 0-1 alkyl R-groups fit produce incorrect results.  This is not published yet, but I was going to publish this in my other paper with the actual fits.  **Therefore, if existing citations exist, lets use them.**

**Comment by bc118** [comment]
Hello!
Thank you for your review and comments! We will complete all of these that are feasible to completed. We started a issue checklist with all the reviewer comments, so we can address them and write our teams notes here [#101](https://github.com/GOMC-WSU/MoSDeF-dihedral-fit/issues/101). Feel free to review this checklist also.

**Comment by bc118** [comment]
@RiesBen Hello!  I hope you are doing well.  We are working to finish this,  have made a lot of progress, and are getting close to completing it.  I do have the following question (s):

**_For this comment "Documentation: Repository Doc strings - API Reference Docs missing? (can be generated with sphinx-apidoc)":_**

- The user's  main functions existed but are **_not labeled as functions but  [data structures](https://mosdef-dihedral-fit.readthedocs.io/en/latest/topic_guides/data_structures.html)_** for the software, and they cover all the possible combinations of variables that the user can enter,  checking them and providing very informative error outputs.  The main functions are (i.e., data structures) .

- I added all the **_[additional utilities](https://mosdef-dihedral-fit.readthedocs.io/en/latest/topic_guides/additional_utilities.html)_**, which are not intended to be used for the direct application of this software.  Is this what you were suggesting here or did you only want the main user function (i.e., data structures in the above)?  

It would be much appreciated if you could clarify this so I can make sure it is what you are suggesting  Can you or let me know if adding the _**additional utilities**_ what you were thinking or simply keeping the _**data structure**_s  (i.e., main functions)?  

Thank you again for your time on your review and your comments that made this software package better!

**Comment by RiesBen** [comment]
Hi @bc118 ,
a right, this is the section I was searching for! cool, that you added the additional utilities! :) 
I think both in something like API reference or so would be the correct section, no?
Why is it actually Datastructure? Isn't this more a code structure or API reference section?

**Comment by bc118** [comment]
Hi!  Ah, yes.  In the docs titles/labeling, we have tried to keep it consistent across the MoSDeF software family (mbuild, foyer, gmso, forcefield-utilities, and MoSDeF-GOMC, etc.) and Data structures was what we have been using for this section.  I was not around for the original naming of the sections, but I suspect it means how they need to structure the input data and what to expect the output or attributes in the classes. 

I also added a few other features (force to k0=0 for OPLS form or a constant) and a lot more examples.
